### ISSUE TITLE
document is not defined

### ISSUE THREAD

**Issue by leonardo2204** [leefsmp/Re-Flex#22]
Hello there, thanks for the great lib !
I've just started using React but I'm getting the following error:
```
/Users/leonardo2204/Moobie/backoffice-web/node_modules/react-reflex/dist/commonjs/ReflexSplitter.js:270
  document: document };
            ^
ReferenceError: document is not defined
    at Object.<anonymous> (/Users/leonardo2204/Moobie/backoffice-web/node_modules/react-reflex/dist/commonjs/ReflexSplitter.js:270:13)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/leonardo2204/Moobie/backoffice-web/node_modules/react-reflex/dist/commonjs/ReflexContainer.js:43:23)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)

```

I guess this may be related to server side rendering, but I'm not sure how to achieve that.

Thanks

**Comment by leonardo2204** [comment]
```
import 'react-reflex/styles.css'

// then you can import the components
import {
  ReflexContainer,
  ReflexSplitter,
  ReflexElement
} from 'react-reflex'

render() {
    return (
      <div>
        <div>
          <ReflexContainer orientation="vertical">
            <ReflexElement className="left-pane">
            <ExifOrientationImg
            src={this.state.currentPhoto.url}
            height={500} />
            </ReflexElement>

            <ReflexElement className="right-pane">
            <Card>
              <CardText >
                <Card>
                  <CardHeader title={this.state.fullName} />
                  <CardText>
                    <Table selectable={false}>
                      <TableBody displayRowCheckbox={false} showRowHover={false} stripedRows={false}>
                        <TableRow>
                          <TableRowColumn>ID</TableRowColumn>
                          <TableRowColumn className={s.notAllowed}>
                            <CopyableLabel>{this.state.userId}</CopyableLabel>
                          </TableRowColumn>
                        </TableRow>
                        <TableRow>
                          <TableRowColumn>Nome</TableRowColumn>
                          {this.renderEditableField('fullName')}
                        </TableRow>
                        <TableRow>
                          <TableRowColumn>CPF</TableRowColumn>
                          <TableRowColumn className={s.notAllowed}>
                            <CopyableLabel>{this.state.cpf.replace(/(\d{3})(\d{3})(\d{3})(\d{2})/g, "\$1.\$2.\$3\-\$4")}</CopyableLabel>
                          </TableRowColumn>
                        </TableRow>
                        <TableRow>
                          <TableRowColumn>Data de Nascimento</TableRowColumn>
                          {this.renderEditableField('birthDate')}
                        </TableRow>
                        <TableRow>
                          <TableRowColumn>Email</TableRowColumn>
                          {this.renderEditableField('email', true)}
                        </TableRow>
                        <TableRow>
                          <TableRowColumn>Telefone</TableRowColumn>
                          {this.renderEditableField('mobile', true)}
                        </TableRow>
                        <TableRow>
                          <TableRowColumn>Nro. Registro CNH</TableRowColumn>
                          {this.renderEditableField('cnhNumber')}
                        </TableRow>
                        <TableRow>
                          <TableRowColumn>Cod. Segurança CNH</TableRowColumn>
                          {this.renderEditableField('securityCode')}
                        </TableRow>
                        <TableRow>
                          <TableRowColumn>Vencimento CNH</TableRowColumn>
                          {this.renderEditableField('expireDate')}
                        </TableRow>
                        <TableRow>
                          <TableRowColumn>Imagem da CNH</TableRowColumn>
                          <TableRowColumn>
                            <Chip style={{ float: "left", marginLeft: "5px" }}
                              onTouchTap={() => this.handlePhotoClick("Foto da CNH", this.getPrivateUrl(this.state.cnhPhoto), this.state.cnhPhoto, 'cnh')}>
                              <Avatar src={this.getPrivateUrl(this.state.cnhPhoto)} />
                              Foto da CNH
                          </Chip>

                          </TableRowColumn>
                        </TableRow>
                        <TableRow>
                          <TableRowColumn>Fotos</TableRowColumn>
                          <TableRowColumn>
                            <Chip style={{ float: "left", marginLeft: "5px" }}
                              onTouchTap={() => this.handlePhotoClick("Foto de perfil", this.state.profilePicture, this.state.profilePicture, 'profile', true)}>
                              <Avatar src={this.state.profilePicture} />Foto de perfil
                          </Chip>
                          </TableRowColumn>
                        </TableRow>
                        <TableRow>
                          <TableRowColumn>Facebook</TableRowColumn>
                          <TableRowColumn>
                            {this.state.facebookId ?
                              <a target="blank"
                                href={`http://facebook.com/${this.state.facebookId}`}>{this.state.facebookId}</a>
                              : !this.state.editedFields.includes('facebook')
                                ? <span className={s.inputLabel}
                                  onDoubleClick={() => this.handleFieldDoubleClick('facebook')}>
                                  {
                                    this.state.facebookUrl ?
                                      <a target="blank"
                                        href={this.state.facebookUrl}>{this.state.facebookUrl}</a> : 'Adicionar facebook'
                                  }
                                </span>
                                : <TextField
                                  hintText="Adicionar facebook"
                                  value={this.state.facebookUrl}
                                  onChange={(event) => this.setState({ facebookUrl: event.target.value })}
                                  id="facebook" />
                            }
                          </TableRowColumn>
                        </TableRow>
                        <TableRow>
                          <TableRowColumn />
                          <TableRowColumn style={{ textAlign: 'right' }}>
                            {this.state.editing &&
                              (<RaisedButton label="Salvar" primary={true}
                                onTouchTap={this.handleUserDataSubmit} />)
                            }
                          </TableRowColumn>
                        </TableRow>
                      </TableBody>
                    </Table>
                  </CardText>
                </Card>
              </CardText>
              </Card>
              </ReflexElement>
              </ReflexContainer>
```
I'm using it along with material-ui !

**Comment by leefsmp** [comment]
The lib has never been tested for server-side rendering actually... you can try as a quick workaround to pass a document property to the `ReflexSplitter`, but it seems your code is not using any splitter at all... 

    const document = {
        removeEventListener: () => {},
        addEventListener: () => {}
    }

    <ReflexSplitter document={document}>

**Comment by leefsmp** [comment]
I'm assuming this solves your issue, so I will go ahead and close that ticket. Feel free to speak up if otherwise.

**Comment by george-norris-salesforce** [comment]
Having same issue. Using Re-Flex with Next.js. 
Adding `<ReflexSplitter document={document}>` didn't fix.

**Comment by stoplion** [comment]
Having same issue. Using Re-Flex with Next.js. Adding <ReflexSplitter document={document}> didn't fix.

**Comment by leefsmp** [comment]
By "seem to" fix, I assume you mean Adding `<ReflexSplitter document={document}>` fixes the issue. Thanks to clarify otherwise.

**Comment by ampsarfraz** [comment]
@stoplion hi .. can you please let me know how you fixed this issue with next.js?

**Comment by stoplion** [comment]
I didn't end up fixing it. I switched it up and am using a Express/Webpack setup. @ampsarfraz

**Comment by smeijer** [comment]
Is it an idea to embed this stub in this project, instead of every place where it's called? SSR gets more common by the day. A simple `document = document || stub` or `document = typeof window === 'undefined' ? stub : document` would do.

Awesome library! But giving me some troubles on SSR as well.

**Comment by leefsmp** [comment]
If you have a suggestion for a fix I'm happy to hear it. If providing a stub for document doesn't help then what is the issue and how to fix it... or is the problem that the stub is missing in some modules?

**Comment by smeijer** [comment]
I've noticed the stub mentioned before doesn't work, as the `ReflexContainer` also includes the `ReflexSplitter` file. It doesn't use it but still, it makes a call to `document`. Causing "compile time" issues.

It's like running `var a = { b }` in your console. That also throws a `b is not defined` error. Unless `b` is defined of course.

The assignment of `defaultProps` happens as soon as the splitter file is included somewhere. It doesn't wait for the class to be initiated.

I was able to fix the issues, and a PR will follow in a few minutes. I did have a few problems getting your build chain running though. So rebuilding the project and bumping the version number, is up to you.
